tests/e2e/tests/pilot/util/environment.go

```diff
@@ -163,18 +174,55 @@ func (e *Environment) ToTemplateData() TemplateData {
 		CABundle:               e.CABundle,
 		RDSv2:                  e.Config.RDSv2,
 		ImagePullPolicy:        e.Config.ImagePullPolicy,
+		MultiCluster:           e.Config.ClusterRegistriesDir,
 	}
 }
 
 // Setup creates the k8s environment and deploys the test apps
 func (e *Environment) Setup() error {
-	if e.Config.KubeConfig == "" {
+	var err error
+	if e.Config.KubeConfig != "" {
+		if _, e.KubeClient, err = kube.CreateInterface(e.Config.KubeConfig); err != nil {
+			return err
+		}
+	}
+	// ClusterRegistiresDir indicates the Kubernetes cluster config should come from files versus KUBECONFIG
+	// environmental variable.  The test config can be defined to use either a single cluster or 2 clusters
+	if e.Config.ClusterRegistriesDir != "" {
+		e.clusterStore, err = clusterregistry.ReadClusters(e.Config.ClusterRegistriesDir)
+		if e.clusterStore == nil {
+			log.Errorf("Failed to clusters in the ClusterRegistriesDir %s\n", e.Config.ClusterRegistriesDir)
```

Comment:
if err == nil then the tests will still try to run.  IMO, you should set err = fmt.Errorf(...) here